certification/internal/engine/engine_test.go

```diff
@@ -1,10 +1,111 @@
 package engine
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http/httptest"
+	"net/url"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/random"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
 )
 
+var _ = Describe("Execute Checks tests", func() {
+	var src string
+	var engine CraneEngine
+	BeforeEach(func() {
+		// Set up a fake registry.
+		s := httptest.NewServer(registry.New())
+		DeferCleanup(func() {
+			s.Close()
+		})
+		u, err := url.Parse(s.URL)
+		Expect(err).ToNot(HaveOccurred())
+
+		src = fmt.Sprintf("%s/test/crane", u.Host)
```

Comment:
Is this a built-in "image" that the register package makes available for use in tests? I would benefit from a comment here.

Reply:
Based on his other PR's, I understand that you can place any image into the registry then look for it at a later time. So I see this as create fake registry, give the image a src, create the random image, then push it to the fake registry, so the fake registry can have N number of images. That's my understanding at least. I might be wrong.